kartothek/core/dataset.py

```diff
@@ -376,29 +376,36 @@ def get_indices_as_dataframe(
         Parameters
         ----------
         """
+        if not self.primary_indices_loaded:
+            # self.load_partition_indices is not inplace
+            dm = self.load_partition_indices()
+        else:
+            dm = self
+
         if columns is None:
-            columns = sorted(self.indices.keys())
-        elif columns == []:
-            return pd.DataFrame(index=self.partitions)
+            columns = sorted(dm.indices.keys())
+
+        if columns == []:
+            return pd.DataFrame(index=dm.partitions)
+
+        if set(columns).difference(dm.indices):
+            # TODO: What to do here? Why not just subset?
+            raise ValueError("Columns not in partition_keys.")
```

Comment:
I think this is a matter of API design. I consider it a bit more user friendly to raise if the input is unexpected. If I compare this to pandas API, I usually receive a KeyError if I'm trying to perform an operation on something which isn't there. Why should we choose a subset here?

Reply:
Should probably be something like

`Requested indices for columns '{set(columns).difference(dm.indices)}' which were not indexed.`

since this does not only affect partition columns